halo/attest/keeper/valcache.go

```diff
@@ -0,0 +1,87 @@
+package keeper
+
+import (
+	"context"
+	"sync"
+
+	vtypes "github.com/omni-network/omni/halo/valsync/types"
+	"github.com/omni-network/omni/lib/errors"
+
+	"github.com/cometbft/cometbft/crypto"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// valCache is a simple read-through cache for validator eth-to-comet address lookups.
+type valCache struct {
+	sync.RWMutex
+	valsByAddr map[common.Address]*vtypes.Validator
```

Comment:
why not just have this be a `map[common.Address]crypto.Address`?

Reply:
agreed, will limit this to the use-case it should be used in.